tests/integration/env.bats

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bats
+# shellcheck disable=SC2016
+# This disables the check for shell variables inside single quotes
+# We do that all the time in this file, as we are testing env vars.
+
+load helpers
+
+function setup() {
+	setup_busybox
+}
+
+function teardown() {
+	teardown_bundle
+}
+
+@test "non-empty HOME env is used" {
+	update_config ' .process.env += ["HOME=/override"]'
+	update_config ' .process.args += ["-c", "echo $HOME"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == '/override' ]]
+}
+
+@test "empty HOME env var is overridden" {
+	update_config ' .process.env += ["HOME="]'
+	update_config ' .process.args += ["-c", "echo $HOME"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == '/root' ]]
+}
+
+@test "empty HOME env var is overridden with multiple overrides" {
+	update_config ' .process.env += ["HOME=/override", "HOME="]'
+	update_config ' .process.args += ["-c", "echo $HOME"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == '/root' ]]
+}
+
+@test "env var HOME is set only once" {
+	# Run env without a shell to print all the env vars, including if one is set multiple times.
+	update_config ' .process.args = ["env"]'
+	update_config ' .process.env = ["HOME=", "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	# There should be 2 words/env-vars: HOME and PATH.
+	var_count=$(echo "${lines[*]}" | wc -w)
+	[[ $var_count -eq 2 ]]
+}
+
+@test "env var with new-line is honored" {
+	update_config ' .process.env += ["NEW_LINE_ENV=\n"]'
+	update_config ' .process.args += ["-c", "test -n \"$NEW_LINE_ENV\"; echo result: \"$?\""]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	# test returns 0 if the variable is set
+	[[ "${lines[0]}" == *"result: 0"* ]]
+}
+
+@test "env var override" {
+	update_config ' .process.env += ["ONE=two", "ONE=three"]'
+	update_config ' .process.args += ["-c", "echo ONE=\"$ONE\""]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == *"ONE=three"* ]]
+}
+
+@test "empty env var makes it unset" {
+	update_config ' .process.env += ["ONE=", "ONE=two", "ONE="]'
+	update_config ' .process.args += ["-c", "test -n \"$ONE\"; echo result: \"$?\"; echo ONE=X\"$ONE\"X"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+	# test returns 1 if the variable is not set.
```

Comment:
This is not entirely correct. `test -n` checks if the string is empty, it's `test -v` that checks if the variable is set.

And, of course, empty value does not make a variable unset, and it shouldn't.

Reply:
Oh, is it a built-in? I don't see `-v` in my test manpage.

But with the env duplication thing we discussed, I removed this test. I'm testing it's only set once and then that is set to the right value now (see "empty env var override is set only once" and "env var override")